test/common/schemaform/ArrayField.unit.spec.jsx

```diff
@@ -196,4 +196,42 @@ describe('Schemaform <ArrayField>', () => {
       expect(onBlur.calledWith([0, 'path'])).to.be.true;
     });
   });
+  it('should disable add when data has not been changed', () => {
+    const idSchema = {};
+    const schema = {
+      type: 'array',
+      items: {
+        type: 'object',
+        properties: {
+          field: {
+            type: 'string'
+          }
+        }
+      }
+    };
+    const uiSchema = {
+      'ui:title': 'List of things',
+      'ui:options': {
+        viewField: f => f
+      }
+    };
+    const errorSchema = {};
+    const onChange = sinon.spy();
+    const onBlur = sinon.spy();
+    const tree = SkinDeep.shallowRender(
+      <ArrayField
+          schema={schema}
+          errorSchema={errorSchema}
+          uiSchema={uiSchema}
+          idSchema={idSchema}
+          registry={registry}
+          onChange={onChange}
+          onBlur={onBlur}
```

Comment:
I was wondering why these were necessary in this test, so looked in `ArrayField`'s props to see if they were required. `onChange` is, but `onBlur` never made it into `ArrayField.propTypes` at all. Just so you're aware.